### PR TITLE
SafeCash (Testnet) Added

### PR DESCRIPTION
**Pull request by hashtobewild**
This commit adds SafeCash (Testnet) support and also some minor changes to messages to make debugging easier.

### CODE REVIEW

File: coins/safecash.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+            "N": 192,
+            "K": 7,
+            "personalization": "SafeCash"
+    },
+    "requireShielding": false,
+    "txfee": 0.0001,
+    "peerMagic": "5a45524f",
+    "payFoundersReward": true,
+    "payAllFounders": true,
+    "foundersRewardAddresses": ["SPfqCriDDxsaH854z3YHigCNUPfSdN3sr1","SaSmUn7LkjmV2WUH1dKFK1xZR1FMRT4qrn","SbfnZN1rPdFq4WsfUUjxphGFcfEits7JXH","SSW62dCjdCxNbKiZdmJh5GLepauY1J1rXU","SW9GN4tZVfPftAQvGmZQLbK6NqePSs8fX1"],
```
Comment (egyptianbman) [inline coins/safecash.json]: Can you fix the alignment in this file? We've put an https://editorconfig.com [file](https://github.com/s-nomp/s-nomp/blob/master/.editorconfig) in the repo that should help with styling if your IDE supports it. Primarily, four space indentation, expand arrays and objects (i.e. lines 18-22 not being on one line), etc. Feel free to check out the other configs for ideas.

File: pool_configs/examples/safecash.json [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+    "tlsOptions": {
+        "enabled": false,
+        "serverKey": "",
+        "serverCert": "",
+        "ca": ""
+    },
+
+    "ports": {
+        "5654": {
```
Comment (egyptianbman) [inline pool_configs/examples/safecash.json]: Can you set this to 1234?

File: pool_configs/examples/safecash.json [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+        "disableTransactions": true
+    },
+
+    "mposMode": {
+        "enabled": false,
+        "host": "127.0.0.1",
+        "port": 3306,
+        "user": "me",
+        "password": "mypass",
+        "database": "zer_testnet",
```
Comment (egyptianbman) [inline pool_configs/examples/safecash.json]: Can you set this to scash_testnet?

**Review by egyptianbman — CHANGES_REQUESTED** [review]
Thank you for providing this. I haven't had a chance to verify the ports are all correct yet but I left a couple of requests for now.

**Comment by Gumbees** [comment]
@egyptianbman all configs tested. Merge after https://github.com/s-nomp/node-stratum-pool/pull/12

File: coins/genesis.json [hunk 4]
```diff
@@ -0,0 +1,35 @@
+{
+    "name": "genesis",
+    "symbol": "GENX",
+    "algorithm": "equihash",
+    "parameters": {
+            "N": 192,
```
Comment (egyptianbman) [inline coins/genesis.json]: Can you fix the alignment here? Should be four spaces per tab

File: coins/testnet/genesis.json [hunk 5]
```diff
@@ -0,0 +1,35 @@
+{
+    "name": "genesis_testnet",
+    "symbol": "GENXT",
+    "algorithm": "equihash",
+    "parameters": {
+            "N": 192,
```
Comment (egyptianbman) [inline coins/testnet/genesis.json]: Same here with the spaces

File: coins/genesis.json [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+    ],
+    "infrastructureAddresses": [
+        "SYSaqv2yChvf462gqmVX8VLrh3m6mXiPLd"
+    ],
+    "giveawayAddresses": [
+        "SQnWFjgxMJyyMQK97gjdF8gYBUUH3kSWEA"
+    ],
+    "explorer": {
+        "txURL": "http://explorer.genesisnetwork.io/tx",
+        "blockURL": "http://explorer.genesisnetwork.io/block",
```
Comment (egyptianbman) [inline coins/genesis.json]: I believe these need the trailing slash.

File: coins/testnet/genesis.json [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+    ],
+    "infrastructureAddresses": [
+        "cFDCRyLcXFWCPMMyxUJajGCPdJLoyAzck6"
+    ],
+    "giveawayAddresses": [
+        "c4NjsSBF4NGXbzxAJUiK1P1KfmxpxXiaaP"
+    ],
+    "explorer": {
+        "txURL": "https://testnet.genesisnetwork.io/tx",
+        "blockURL": "https://testnet.genesisnetwork.io/block",
```
Comment (egyptianbman) [inline coins/testnet/genesis.json]: I believe these need the trailing slash.

File: libs/paymentProcessor.js [hunk 8]
```diff
@@ -98,7 +98,7 @@ function SetupForPool(logger, poolOptions, setupFinished){
     function validateAddress (callback){
         daemon.cmd('validateaddress', [poolOptions.address], function(result) {
             if (result.error){
-                logger.error(logSystem, logComponent, 'Error with payment processing daemon ' + JSON.stringify(result.error));
+                logger.error(logSystem, logComponent, 'Error with payment processing daemon (validateAddress) ' + JSON.stringify(result.error));
```
Comment (egyptianbman) [inline libs/paymentProcessor.js:101]: Thank you for these updates!

File: pool_configs/examples/genesis.json [hunk 9]
```diff
@@ -0,0 +1,90 @@
+{
+    "enabled": true,
```
Comment (egyptianbman) [inline pool_configs/examples/genesis.json]: Default enabled to `false`

File: pool_configs/examples/genesis.testnet.json [hunk 10]
```diff
@@ -0,0 +1,90 @@
+{
+    "enabled": true,
```
Comment (egyptianbman) [inline pool_configs/examples/genesis.testnet.json]: Default enabled to `false`

**Review by egyptianbman — CHANGES_REQUESTED** [review]
A few last minor requests; thanks!

**Review by egyptianbman — APPROVED** [review]
Thank you!

**Pull request merged**